earth_enterprise/src/fusion/asset-tests/bin/generate_test_asset_xmls.py

```diff
@@ -417,18 +480,63 @@ def generate_invalid_raster_projects(self):
             inset, self.invalid_raster_product_asset_version)
         raster_project_asset.output(self.asset_root, self.xml_writer)
 
+        self.progress_meter.advance(1)
+
+class TestAssetGenerator(TestInvalidAssetGenerator):
+    def __init__(
+        self, output_valid_raster_projects=True,
+        output_invalid_raster_projects=True
+    ):
+        super(TestAssetGenerator, self).__init__()
+        self.output_valid_raster_projects = output_valid_raster_projects
+        self.output_invalid_raster_projects = output_invalid_raster_projects
+
+    def create_common_assets(self):
+        self.create_valid_raster_product_asset()
+        self.create_invalid_raster_product_asset()
+
+    def write_common_assets(self):
+        self.valid_raster_product_asset.output(self.asset_root, self.xml_writer)
+        self.invalid_raster_product_asset.output(
+            self.asset_root, self.xml_writer)
+
+        self.progress_meter.advance(2)
+
     def generate_raster_projects(self):
-        self.generate_valid_raster_projects()
-        self.generate_invalid_raster_projects()
-    
+        if self.output_valid_raster_projects:
+            self.generate_valid_raster_projects()
+        if self.output_invalid_raster_projects:
+            self.generate_invalid_raster_projects()
+
     def generate_assets(self):
+        if self.output_valid_raster_projects:
+            self.progress_meter.end_value += self.valid_raster_project_count
+        if self.output_invalid_raster_projects:
+            self.progress_meter.end_value += self.invalid_raster_project_count
+
         self.create_common_assets()
         self.generate_raster_projects()
         self.write_common_assets()
 
+        self.progress_meter.close()
+
 
 def main():
-    generator = TestAssetGenerator()
+    parser = argparse.ArgumentParser(
+        description="Generate Fusion asset XMLs for testing.")
+
+    parser.add_argument(
```

Comment:
Minor: I prefer "positive" arguments like `--generate-valid-raster-projects`. It makes it easier to reason about the code that uses them. However, that's more of a style question.